core/src/main/java/hudson/security/ACL.java

```diff
@@ -91,6 +113,26 @@ public final boolean hasPermission(@Nonnull Permission p) {
         return hasPermission(a, p);
     }
 
+    /**
+     * Checks if the current security principal has any of the permissions.
+     *
+     * @return false
+     *      if the user doesn't have one of the required permissions.
+     */
+    public final boolean hasAnyPermission(@Nonnull Permission... permissions) {
```

Comment:
Not sure this would work in the case that there are no Permissions?   (`hasAnyPermission()`)
in this case permissions is non null but has a length of zero.

thus it think it will return `false` whereas logically you may expect it to do the oposite.
I'm fine with false - but this should probably be added to the javadoc if this is intentional (and maybe covered by a unit test to make sure it does not change and introduce a securioty issue in some plugin in the future if it changes?)